tchannel/tornado/tchannel.py

```diff
@@ -221,7 +229,16 @@ def listen(self, port=None):
 
         :returns:
             Returns immediately.
+
+        :raises AlreadyListeningError:
+            If listen was already called.
         """
+
+        if self.is_listening():
+            raise AlreadyListeningError(
```

Comment:
It is weird to call AlreadyListeningError. It could be ServerError, or ServerInitError.